### PR TITLE
fix(docs): directory sync fix + test bed rewrites

### PR DESCRIPTION
**Pull request by kanadgupta**
## 🧰 Changes

Another follow-up to #410. This time we're focusing on our `docs` commands:
- [x] Fixed a couple bugs with our logic for syncing markdown directories:
   1. It previously was returning a successful exit code (`0`) if the docs syncing failed for any reason. Now a failure rejects the promise and returns a failure exit code (`1`).
   2. If one doc was invalid and failed, it would previously continue with the rest, and print a JSON object of the docs that were successful. Now, when a doc fails, it exits the entire process with a failure exit code (`1`) and returns an error message containing the problematic file path.
- [x] Replacing all `console.log` instances with `console.info`.
- [x] Rewriting our tests to check for the function outputs (i.e. Promise resolution/rejection) rather than checking for console outputs.
- [x] Makes our version prompt (see #344) more generic so the language makes sense in more places.

## 🧬 QA & Testing

See tests.

### CODE REVIEW

File: src/lib/prompts.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ exports.generatePrompts = (versionList, selectOnly = false) => [
   {
     type: 'select',
     name: 'option',
-    message: 'Would you like to use an existing version or create a new one to associate with your OAS file?',
```
Comment (kanadgupta) [inline src/lib/prompts.js:9]: We were previously using this language in a bunch of places (version commands, docs commands, etc.) 😬

File: src/cmds/docs/index.js [hunk 2]
```diff
@@ -48,7 +48,10 @@ exports.run = async function (opts) {
     return Promise.reject(new Error(`No folder provided. Usage \`${config.cli} ${exports.usage}\`.`));
   }
 
-  const selectedVersion = await getProjectVersion(version, key, true).catch(e => {
+  // TODO: should we allow version selection at all here?
+  // Let's revisit this once we re-evaluate our category logic in the API.
+  // Ideally we should ignore this parameter entirely if the category is included.
+  const selectedVersion = await getProjectVersion(version, key, false).catch(e => {
     return Promise.reject(e);
```
Comment (erunion) [inline src/cmds/docs/index.js:55]: Do we need to handle this rejection here? I think this might cause the same exit code issue you fixed last month.

**Review by erunion — APPROVED** [review]

File: src/cmds/docs/index.js (hunk 2 above)
Comment (kanadgupta) [reply src/cmds/docs/index.js:55]: Oh yep you're right, thanks!

File: __tests__/cmds/docs.test.js [hunk 3]
```diff
@@ -363,55 +346,55 @@ describe('rdme docs:edit', () => {
...
-      message: "We couldn't save this doc ({error})",
+      message: `We couldn't save this doc (${slug})`,
       suggestion: '...a suggestion to resolve the issue...',
       help: 'If you need help, email support@readme.io and mention log "fake-metrics-uuid".',
-    });
+    };
+
+    const getMock = nock(config.host).get(`/api/v1/docs/${slug}`).reply(200, { body });
+
+    const putMock = nock(config.host).put(`/api/v1/docs/${slug}`).reply(400, errorObject);
```
[the remark this answers is not in the document]
Comment (kanadgupta) [reply __tests__/cmds/docs.test.js:397]: whoa TIL!

[the remark this answers is not in the document]
Comment (erunion) [reply __tests__/cmds/docs.test.js:397]: I've sometimes had trouble with it in the past so YMMV

File: __tests__/cmds/docs.test.js [hunk 4]
```diff
@@ -363,55 +346,55 @@ describe('rdme docs:edit', () => {
...
+    await expect(docsEdit.run({ slug, key, version: '1.0.0', mockEditor })).resolves.toBeUndefined();
 
-      expect(console.log).toHaveBeenCalledWith('Doc successfully updated. Cleaning up local file.');
-    });
+    getMock.done();
+    putMock.done();
+    versionMock.done();
+
+    expect(fs.existsSync(`${slug}.md`)).toBe(false);
+    await expect(console.info).toHaveBeenCalledWith('Doc successfully updated. Cleaning up local file.');
```
[the remark this answers is not in the document]
Comment (kanadgupta) [reply __tests__/cmds/docs.test.js:356]: Nope, thank you!

File: __tests__/cmds/docs.test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kanadgupta) [reply __tests__/cmds/docs.test.js:397]: Ok looking through this, there's a lot of nock separation like this and I'm worried the nocks will become less readable if we lump them together. Thoughts?

File: src/cmds/docs/index.js (hunk 2 above)
Comment (kanadgupta) [reply src/cmds/docs/index.js]: fixed in https://github.com/readmeio/rdme/pull/421/commits/fe257a83c15816a5178aacebf1d5207c02975b3a

File: __tests__/cmds/docs.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (kanadgupta) [reply __tests__/cmds/docs.test.js]: fixed in https://github.com/readmeio/rdme/pull/421/commits/fe257a83c15816a5178aacebf1d5207c02975b3a

File: __tests__/cmds/docs.test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (erunion) [reply __tests__/cmds/docs.test.js:397]: Collapsing everything might be a bit gross so I'm cool with whatever you're cool with.

**Pull request merged**